src/services/CalcFactor.jl

```diff
@@ -0,0 +1,186 @@
+# New factor interface, something perhaps like this
+
+
+export calcFactorResidualTemporary
+
+
+# Helper function to construct CF from a CCW
+CalcFactor(ccwl::CommonConvWrapper) = CalcFactor( ccwl.usrfnc!, _getFMdThread(ccwl), 0, length(ccwl.measurement), ccwl.measurement, ccwl.params)
+
+
+
+"""
+    $SIGNATURES
+
+Sample the factor stochastic model `N::Int` times and store the samples in the preallocated `ccw.measurement` container.
+
+DevNotes
+- Use in place operations where possible and remember `measurement` is a `::Tuple`.
+- TODO only works on `.threadid()==1` at present, see #1094
+- Also see, JuliaRobotics/RoME.jl#465
+"""
+function sampleFactor(cf::CalcFactor{<:AbstractFactor}, 
+                      N::Int=1  )
+  #
+  getSample(cf, N)
```

Comment:
Hi @Affie, here is where getSample gets called from IIF, not sure if there are any other `rand` calls still floating around inside IIF -- they should all come here instead.